src/App.tsx

```diff
@@ -15,36 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortField {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const getPreparedGoods = (
+  listOfgoods: string[],
+  sortField: SortField | '',
+  isReversed: boolean,
+): string[] => {
+  const preparedGoods = [...listOfgoods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortField.ALPHABET:
+          return a.localeCompare(b);
+        case SortField.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortField.ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SortField.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
-
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
```

Comment:
We can write it in one quotation marks.
```suggestion
          className={cn('button is-success', {
```